### DISCUSSION TITLE
How is SYCL support now days?

### DISCUSSION THREAD

**Discussion by bertsons** [ggml-org/llama.cpp#10392] [General]
I'm just wondering how stable SYCL support is now days with Llama.cpp; primarily for Intel GPUs (I'm not bothered about the extensions for AMD/NVIDIA)?

I'm curious because the SYCL README (https://github.com/ggerganov/llama.cpp/blob/master/docs/backend/SYCL.md) is still recommending a build from a release back in May; b3038? I presume this means that SYCL is currently unstable?

**Comment by NeoZhangJianyu** [comment]
The SYCL backend is still supported now!
The stable release is not updated for a long time. 
I will update it with latest stable release.

**Reply by bertsons** [reply]
Thanks! I am interested in SYCL support because it apparently runs much better than vulkan on Intel GPUs, and I would like to include SYCL support in a downstream open source project I am working on for Llama.cpp integration with unreal engine.

**Reply by NeoZhangJianyu** [reply]
It's updated by https://github.com/ggerganov/llama.cpp/pull/10395.

Could I know what's the Intel GPU in your project?
What's " unreal engine"?

**Reply by bertsons** [reply]
Unreal Engine is a popular game engine. My Laptop has an Intel Iris Xe GPU. It's nothing serious, but is useful to prove SYCL compatibility.

**Reply by bertsons** [reply]
> It's updated by #10395.

By the way, what is your process to updating the stable release? I am wondering because I am interested in keeping it up to date.

**Reply by NeoZhangJianyu** [reply]
OK, it's clear to me!
If you use Meteor Lake or newer Intel CPU in Laptop, the built-in Arc GPU will be more powerful.
The iGPU in 11th Core CPU or newer are supported by SYCL.
But the performance is not good as built-in Arc GPU.

In general, I will test the package with Arc 770 (Linux) and MTL built-in GPU (Windows) to confirm the quality.
The running time CI of SYCL is still in development.
We can't confirm the quality of the PRs before merge.

If you have any issue, please report issue with [SYCL] as prefix in title. we will track and support.
